### PR TITLE
Update and cleanup of #209.

### PR DESCRIPTION
**Pull request by jmchilton**

### CODE REVIEW

File: lib/galaxy/datatypes/tabular.py [hunk 1]
```diff
@@ -691,6 +695,23 @@ def set_meta( self, dataset, **kwd ):
...
             dataset.metadata.sample_names = line.split()[ 9: ]
 
+    @staticmethod
+    def merge(split_files, output_file):
+        tmp_dir = tempfile.mkdtemp()
+        stderr_name = tempfile.NamedTemporaryFile(dir=tmp_dir, prefix="bam_merge_stderr").name
+        command = ["bcftools", "concat"] + split_files + ["-o", output_file]
+        log.info("Merging vcf files with command [%s]" % " ".join(command))
+        proc = subprocess.Popen( args=command, stderr=open( stderr_name, 'wb' ) )
+        exit_code = proc.wait()
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/tabular.py]: You can combine the previous 2 lines using `subprocess.call()` .

File: lib/galaxy/datatypes/tabular.py [hunk 2]
```diff
@@ -691,6 +695,23 @@ def set_meta( self, dataset, **kwd ):
...
+        # Did merge succeed?
+        stderr = open(stderr_name).read().strip()
+        if stderr:
+            if exit_code != 0:
+                shutil.rmtree(tmp_dir)  # clean up
```
Comment (bgruening) [inline lib/galaxy/datatypes/tabular.py]: Both if's can be combined  and shutil.rmtree() can be used instead of os.unlink() at the bottom.

**Comment by jmchilton** [comment]
Updated based on comments.

**Comment by bgruening** [comment]
:+1:

File: lib/galaxy/datatypes/tabular.py [hunk 3]
```diff
@@ -691,6 +695,22 @@ def set_meta( self, dataset, **kwd ):
             # Found header line, get sample names.
             dataset.metadata.sample_names = line.split()[ 9: ]
 
+    @staticmethod
+    def merge(split_files, output_file):
+        tmp_dir = tempfile.mkdtemp()
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/tabular.py]: One final comment: does it make sense to create `tmp_dir` only to pass it to `NamedTemporaryFile()` ?

Comment (jmchilton) [inline lib/galaxy/datatypes/tabular.py]: I guess not, fixed.

**Comment by nsoranzo** [comment]
:+1:

**Pull request merged**